src/AzureExtension/Widgets/WidgetProvider.cs

```diff
@@ -33,7 +33,7 @@ private void InitializeWidget(WidgetContext widgetContext, string state)
         var widgetId = widgetContext.Id;
         var widgetDefinitionId = widgetContext.DefinitionId;
         Log.Logger()?.ReportDebug($"Calling Initialize for Widget Id: {widgetId} - {widgetDefinitionId}");
-        if (widgetDefinitionRegistry.ContainsKey(widgetDefinitionId))
+        if (widgetDefinitionRegistry.TryGetValue(widgetDefinitionId, out var _))
         {
             if (!runningWidgets.ContainsKey(widgetId))
             {

```

Comment:
you should make use of the value you're storing in from TryGetValue instead of calling 'var factory = widgetDefinitionRegistry[widgetDefinitionId];' below

Reply:
Nice catch.  Thank you.